test/mitmproxy/proxy/protocol/test_http1.py

```diff
@@ -77,3 +79,35 @@ def test_head_content_length(self):
                 """head:'%s/p/200:h"Content-Length"="42"'""" % self.server.urlbase
             )
         assert resp.headers["Content-Length"] == "42"
+
+
+class Stream:
+
+    def requestheaders(self, f):
+        f.request.stream = True
+
+    def responseheaders(self, f):
+        f.response.stream = True
+
+
+def write(self, v):
+    assert len(v) <= 4096  # Chunk size is 4096
+    if v:
+        self.first_byte_timestamp = self.first_byte_timestamp
+
+        if hasattr(self.o, "sendall"):
+            self.add_log(v)
+            return self.o.sendall(v)
+        else:
+            r = self.o.write(v)
+            self.add_log(v[:r])
+            return r
+
+
+class TestRequestStreaming(tservers.HTTPProxyTest):
+    @mock.patch("mitmproxy.net.tcp.Writer.write", side_effect=write, autospec=True)
```

Comment:
If you move the mocking after creating the pathoc object, we should only affect new connections, but not pathoc...